### ISSUE TITLE
Error "Failed to remove sub" while shutting down clustered Vertx

### ISSUE THREAD

**Issue by nolequen** [eclipse-vertx/vert.x#3209]
```
io.vertx.core.eventbus.impl.clustered.ClusteredEventBus
SEVERE: Failed to remove sub
com.hazelcast.core.HazelcastInstanceNotActiveException: Hazelcast instance is not active!
	at com.hazelcast.spi.AbstractDistributedObject.throwNotActiveException(AbstractDistributedObject.java:105)
	at com.hazelcast.spi.AbstractDistributedObject.lifecycleCheck(AbstractDistributedObject.java:100)
	at com.hazelcast.spi.AbstractDistributedObject.getNodeEngine(AbstractDistributedObject.java:94)
	at com.hazelcast.multimap.impl.ObjectMultiMapProxy.remove(ObjectMultiMapProxy.java:128)
	at io.vertx.spi.cluster.hazelcast.impl.HazelcastAsyncMultiMap.lambda$remove$8(HazelcastAsyncMultiMap.java:169)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:316)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

It seems that the reason is that ```EventBusImpl#unregisterAll``` calls `HandlerRegistration#unregister` with null `doneHandler`. This leads to the cluster leaving before the EventBus is shutted down.

**Comment by tsegismont** [comment]
I will check this before 4.0 but the init/shutdown procedure has been reworked and I believe the issue might be addressed in master already.

**Comment by tsegismont** [comment]
Fixed by #3132 and https://github.com/vert-x3/vertx-hazelcast/pull/126
